### PR TITLE
Add workaround for soundplay file limit

### PR DESCRIPTION
**Pull request by ryanewel**
Signed-off-by: ryanewel <ryanewel@amazon.com>

*Issue #, if available:*
https://github.com/aws-robotics/tts-ros1/issues/26
*Description of changes:*
Instead of passing unique file names to soundplay node this change creates a symlink with a constant name to the output file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tts/scripts/tts_node.py [hunk 1]
```diff
@@ -102,8 +105,16 @@ def do_speak(goal):
 
     if 'Audio File' in r:
         audio_file = r['Audio File']
+        # Soundplay cannot handle more than 32 different file paths.
+        # Creating a symlink to the polly output and then passing that to
+        # to the soundplay node is a workaround.
+        # See: https://github.com/ros-drivers/audio_common/issues/125
+        sym_link_path = os.path.abspath(SOUND_FILE_PATH)
+        os.remove(sym_link_path)
```
Comment (AAlon) [inline tts/scripts/tts_node.py]: Wouldn't this throw an exception the first time? (before the link exists)

File: tts/scripts/tts_node.py [hunk 2]
```diff
@@ -59,6 +60,8 @@
 
 from sound_play.libsoundplay import SoundClient
 
+# Output files will be symlinked to this path
+SOUND_FILE_PATH = os.path.join(os.sep, 'tmp', 'tts_sound_file')
```
Comment (AAlon) [inline tts/scripts/tts_node.py]: Make this platform independent: https://docs.python.org/2/library/tempfile.html#tempfile.tempdir

**Comment by AAlon** [comment]
You didn't explicitly said so, but I assume you verified that this solves the issue? :)

**Comment by ryanewel** [comment]
Yep, should have specified my testing.

I sent voice requests to the TTS node using this script.
```
    for i in `seq 1 $1`; do
        echo $i
        rosrun tts voicer.py 'Hello World'
    done
```

Before this change with 32 requests all but the last would be executed and it would hang on the last one.

After the change it can be run 33 times.

With the second commit fixing the issue with the first run of the node it can be run 33 times on first run and after restarting the node.

File: tts/scripts/tts_node.py (hunk 1 above)
Comment (ryanewel) [reply tts/scripts/tts_node.py]: Turns out it does. Fixed

File: tts/scripts/tts_node.py [hunk 3]
```diff
@@ -102,8 +106,17 @@ def do_speak(goal):
...
+        # Soundplay cannot handle more than 32 different file paths.
+        # Creating a symlink to the polly output and then passing that to
+        # to the soundplay node is a workaround.
+        # See: https://github.com/ros-drivers/audio_common/issues/125
+        sym_link_path = os.path.abspath(SOUND_FILE_PATH)
+        if os.path.lexists(sym_link_path):
+            os.remove(sym_link_path)
         rospy.loginfo('Will play {}'.format(audio_file))
-        play(audio_file)
+        rospy.loginfo('Symlinking {} to {}'.format(audio_file, sym_link_path))
```
Comment (timrobotson) [inline tts/scripts/tts_node.py]: nit: I think this should be a debug statement.

**Comment by nburek** [comment]
Make sure the CI is passing before you merge, but the code looks good to me too.

**Pull request merged**